src/main/tasks_tests_handler.py

```diff
@@ -0,0 +1,212 @@
+
+
+def __separate_in_and_out_files_on_pairs(in_files: list, out_files: list):
+    pairs = []
+    for in_file in in_files:
+        out_file = __get_out_file_by_in_file(in_file)
+        if out_file not in out_files:
+            raise ValueError('List of out files does not contain a file for ' + in_file)
+        pairs.append((in_file, out_file))
+    return pairs
+
+
+def __is_python(extension: str):
+    return extension == 'py'
+
+
+def __get_java_class(source_code: str):
+    class_key_word = 'class'
+    rows = source_code.split('\n')
+    for row in rows:
+        if class_key_word in row:
+            class_index = row.index(class_key_word)
+            return row[class_index + len(class_key_word) + 1:].replace(' ', '').replace('{', '')
+    raise ValueError('Source code does not contain class name!')
+
+
+# Wrap all values from input in the print command
+def __create_py_input_file(txt_in_file: str, task: str, extension='py', file_name=INPUT_FILE_NAME):
+    code = ''
+    with open(TASKS_TESTS_PATH + task + '/' + txt_in_file, 'r') as f:
+        for line in f:
+            code += 'print(' + line.strip('\n') + ')' + '\n'
+    __create_file(code, task, extension, file_name)
+
+
+# For python scripts it is an in file with extension py and for other cases, it is an in file with extension txt
+def __get_in_file_for_current_test(cur_in_file: str, task: str, extension='py'):
+    if __is_python(extension):
+        __create_py_input_file(cur_in_file, task)
+        return INPUT_FILE_NAME + '.' + extension
+    return cur_in_file
+
+
+# Clear not .txt files: we need to clear the generated files (it has not .txt extension)
+def __clear_old_files(task: str):
+    files = next(os.walk(TASKS_TESTS_PATH + task))[2]
+    old_files = list(filter(lambda file_name: '.txt' not in file_name, files))
+    for file in old_files:
+        __remove_file(file, task)
+
+
+def __run_python_test(in_file: str, out: str, task: str, extension='py', source_file_name=SOURCE_FILE_NAME):
+    p1 = Popen(['python', TASKS_TESTS_PATH + task + '/' + in_file], stdout=PIPE)
+    p2 = Popen(['python', TASKS_TESTS_PATH + task + '/' + source_file_name + '.' + extension], stdin=p1.stdout,
+               stdout=PIPE)
+    p1.stdout.close()
+    output = p2.communicate()[0]
+    if output.decode("utf-8").rstrip("\n") == out:
+        return True
+    return False
+
+
+# Run test for compiled languages
+def __run_test(in_file: str, out: str, task: str, popen_args: list):
+    p = Popen(popen_args, stdin=PIPE, stdout=PIPE, stderr=PIPE, universal_newlines=True)
+    cur_out, err = p.communicate(input=__get_content_from_file(in_file, task))
+    if p.returncode == 0 and cur_out.rstrip("\n") == out:
+        return True
+    return False
+
+
+def __compile_program(call_args: list):
+    if call(call_args) != 0:
+        # Error
+        return False
+    return True
+
+
+def __get_compiled_file_path(extension: str, task: str, source_file_name: str):
```

Comment:
The result of this function with several ifs is only used in the `__get_args_for_compiling_program` with the same ifs. Shouldn't we merge it to the `__get_args_for_compiling_program`, counting compiled file path from the right if? 
This would help us to get rid of this function that doesn't look too pretty because of so many ifs.
So, in the case of java it would be something like this:
```
if extension == 'java':
        compiled_file_path = base_path + source_file_name + '.' + extension
        call_args = ['javac', compiled_file_path]
```